runtime/lua/vim/termcap.lua

```diff
@@ -0,0 +1,86 @@
+local M = {}
+
+--- Hex encode a string.
+---
+--- @param str string String to encode
+--- @return string Hex encoded string
+local function hexencode(str)
```

Comment:
Should we put this in `vim.codec.hex.encode`? Then base64 could live in `vim.codec.base64`.

Or `vim.text.hexencode()` ? We definitely will add a `vim.text` module soonish.

Reply:
`vim.text.hex{en,de}code()` sounds fine to me

Reply:
Made a separate PR for `vim.text`, which now blocks this one https://github.com/neovim/neovim/pull/26069